x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/enrollment/CreateEnrollmentTokenTests.java

```diff
@@ -0,0 +1,308 @@
+
+        final MockSecureSettings secureSettings = new MockSecureSettings();
+        secureSettings.setString("xpack.http.ssl.truststore.secure_password", "password");
+        secureSettings.setString("xpack.security.http.ssl.keystore.secure_password", "password");
+        final Settings settings = Settings.builder()
+            .put("xpack.security.enabled", true)
+            .put("xpack.http.ssl.enabled", true)
+            .put( "xpack.security.authc.api_key.enabled", true)
+            .put("xpack.http.ssl.truststore.path", "httpCa.p12")
+            .put("xpack.security.http.ssl.enabled", true)
+            .put("xpack.security.http.ssl.keystore.path", "httpCa.p12")
+            .setSecureSettings(secureSettings)
+            .put("path.home", tempDir)
+            .build();
+        final Environment environment_not_enabled = new Environment(settings, tempDir);
+        final CommandLineHttpClient client = mock(CommandLineHttpClient.class);
+
+        CreateEnrollmentToken createEnrollmentToken = new CreateEnrollmentToken(environment_not_enabled, client);
+        IllegalStateException ex = expectThrows(IllegalStateException.class, () -> createEnrollmentToken.create("elastic",
+            new SecureString("elastic"), new SecureString("password")));
+        assertThat(ex.getMessage(), Matchers.equalTo("'xpack.security.enrollment' must be enabled to create an enrollment token"));
+    }
+
+    public void testReadKeystoreWrongPassword () {
+        Path path = environment.configFile().resolve(environment.settings().get("xpack.security.http.ssl.keystore.path"));
+        SslConfigException ex = expectThrows(SslConfigException.class, () -> CreateEnrollmentToken.readKeyStore(path,
+            new SecureString("wrong_password")));
+        assertThat(ex.getMessage(), Matchers.startsWith("cannot read a [PKCS12] keystore from "));
+    }
+
+    public void testReadKeystoreWrongPath () {
+        Path path = environment.configFile().resolve("");
+        SslConfigException ex = expectThrows(SslConfigException.class, () -> CreateEnrollmentToken.readKeyStore(path,
+            new SecureString("wrong_password")));
+        assertThat(ex.getMessage(), Matchers.startsWith("cannot read a [jks] keystore from "));
+    }
+
+    public void testGetPrivateKeyEntriesWrongPassword () throws Exception {
+        Path path = environment.configFile().resolve(environment.settings().get("xpack.security.http.ssl.keystore.path"));
+        KeyStore keyStore = CreateEnrollmentToken.readKeyStore(path, new SecureString("password"));
+        ElasticsearchException ex = expectThrows(ElasticsearchException.class, () -> CreateEnrollmentToken.getPrivateKeyEntries(keyStore,
+            new SecureString("wrong_password")));
+        assertThat(ex.getMessage(), Matchers.startsWith("failed to list keys and certificates"));
+    }
+
+    public void testGetPrivateKeyEntriesNullKeyStore () {
```

Comment:
readKeystore doesn't return null so how would we end up calling getPrivateKeyEntries with a `null` keystore ?